src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -246,6 +249,15 @@ private static boolean isSameRequester(ConsentArtefact consentDetail, String req
                                                 .thenReturn(consentApprovalResponse(consents)))));
     }
 
+    private Mono<Void> processExpiredConsentRequest(String requestId) {
+        return consentRequestRepository.requestOf(requestId)
+                .filter(consentRequest -> isConsentRequestExpired(consentRequest.getCreatedAt()) &&
+                consentRequest.getStatus().equals(REQUESTED))
+                .switchIfEmpty(Mono.empty())
+                .flatMap(response -> consentRequestRepository.updateStatus(requestId, EXPIRED).
+                        then(Mono.error(ClientError.consentRequestExpired())));
+    }
```

Comment:
send notification to HIU?

there's a queue which is there for that, you may need to enqueue the message though

Reply:
Looking into it, sure.